src/coreclr/jit/importer.cpp

```diff
@@ -7082,16 +7084,29 @@ int Compiler::impBoxPatternMatch(CORINFO_RESOLVED_TOKEN* pResolvedToken,
                     case CEE_BRFALSE_S:
                         if ((nextCodeAddr + ((nextCodeAddr[0] >= CEE_BRFALSE) ? 5 : 2)) <= codeEndp)
                         {
-                            if (makeInlineObservation)
+                            if (flags & MakeInlineObservation)
                             {
                                 compInlineResult->Note(InlineObservation::CALLEE_FOLDABLE_BOX);
                                 return 1 + sizeof(mdToken);
                             }
 
+                            assert(!(flags & IsByRefLike) || ((impStackTop().val->gtFlags & GTF_SIDE_EFFECT) == 0));
+
                             if (!(impStackTop().val->gtFlags & GTF_SIDE_EFFECT))
                             {
```

Comment:
```suggestion
                            if (!(impStackTop().val->gtFlags & GTF_SIDE_EFFECT))
                            {
                                assert((flags & IsByRefLike) == 0);
```
I know this assert was added in my code but I think it is better to just assert inside the if.

Also, now that you're in this function can you update the flag checks to check explicitly against zero? That is the convention we encourage nowadays (https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#1515-use-of-zero).